### PR TITLE
LLaMA-Adapter

### PR DESCRIPTION
**Pull request by awaelchli**
This PR adds a basic implementation of the paper [LLaMA-Adapter: Efficient Fine-tuning of Language Models with Zero-init Attention](https://arxiv.org/abs/2303.16199.pdf) on top of our LLaMA base model. It adds 1) the model definition with adapter logic 2) the script for fine-tuning with the Stanford Alpaca recipe. 

Note:
- No multi-model implementation
- Hyper-parameters are not tuned well (help appreciated)
- Runs on a single GPU with gradient accumulation, which takes longer than the paper which trained on 8 A100 GPUs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -117,7 +117,7 @@ See `python generate.py --help` for more options.
 
 ## Finetune the model
 
-We provide a simple training script in `finetune.py` that instruction-tunes a pretrained model on the [Alpaca](https://github.com/tatsu-lab/stanford_alpaca) dataset and also integrates [LoRA](https://arxiv.org/abs/2106.09685).
+We provide a simple training script in `finetune_lora.py` that instruction-tunes a pretrained model on the [Alpaca](https://github.com/tatsu-lab/stanford_alpaca) dataset and also integrates [LoRA](https://arxiv.org/abs/2106.09685).
```
Comment (lantiga) [inline README.md:120]: We need to feature `finetune_adapter.py` as well in the README (separate section)

**Review by lantiga — APPROVED** [review]
Looks great. Just please add LLaMA-Adapter plus reference (to the paper, not the implementation) to the README

**Review by lantiga — APPROVED** [review]
